src/se/vidstige/jadb/JadbDevice.java

```diff
@@ -126,11 +126,29 @@ public void executeShell(OutputStream output, String command, String... args) th
         return result;
     }
 
-    private int getMode(File file) {
+    public int getMode(File file) {
         //noinspection OctalInteger
         return 0664;
     }
 
+    public void simulateUsbUnplug() throws IOException, JadbException {
+        InputStream stream = executeShell("dumpsys", "battery", "unplug");
+        int i;
+        do{
+            i = stream.read();
+        } while(i != -1);
+        stream.close();
+    }
+
+    public void resetUsbPlug() throws IOException, JadbException {
+        InputStream stream = executeShell("dumpsys", "battery", "reset");
+        int i;
+        do{
+            i = stream.read();
+        } while(i != -1);
+        stream.close();
+    }
```

Comment:
Also the code is duplicated between the two functions above. Please move out to common function.